### DISCUSSION TITLE
Commodore 64 / C64 Cartridge Dumper

### DISCUSSION THREAD

**Discussion by ezcGman** [sanni/cartreader#795] [Ideas & Suggestions]
Hey there,

I was wondering if there's also support for the Commodore / C64 cartridges, but couldn't find any user made adapter for them. Digging through the interwebs a bit, I stumbled over this PoC: https://github.com/segrax/Arduino-C64-Cartridge-Dumper

He basically "just" figured out the pinouts and could successfully serial read the cartridge.

Now: I do have the KiCAD skills to make the adapter for this, but I do not have the skills to write the proper software support for this for the OSCR. Sure, I'm a dev and I do a lot of Arduino / ESP / C++ coding, so that is not the issue; I could port the code in that GitHub repo over to the OSCR here, but that would just be a "dump port" without me actually understanding anything of how cartridges work, how and why it reads what address, etc etc.

So maybe somebody also finds this useful and has more of the software knowledge and can provide the code for it? I would volunteer to do the hardware / KiCAD part :)

Greetings,

Andy!

**Comment by sakman55** [comment]
Hi ezcGman,

I have a test C64 module working with the Cart Reader.  I have a few test carts that are dumping properly.  I'll need help with testing carts with different mapper variations.

If you're interested in helping, message me (skaman) on the Arduino forum.

Take Care!

**Reply by ezcGman** [reply]
Heyhey,

with "module" u mean u built an adapter? I would be a happy to try it out! I don't have THAT many carts, but would be happy to get the PCB printed, soldered and test with it :)

Do you have the Gerbers and the source code for the OSCR somewhere here on GitHub?

Greetings,

Andy!

**Comment by sakman55** [comment]
I'm working on the PCB layout.  My handwired prototype adapter is working with the code.  Preliminary test carts all dump properly.  Note: clean dumps will not match most .CRT "ROMs" since those include a header for each chip.

I need a comprehensive list of C64 carts with the proper sizes and mappers.  Documentation for the various mappers would also be good.  My initial code incorporated a few mappers documented by Luigi Di Fraia.

Still a lot of work to do before a release.
